src/ecalc/libraries/libecalc/common/libecalc/dto/result/results.py

```diff
@@ -134,10 +141,25 @@ class PumpModelResult(ConsumerModelResultBase):
     operational_head: Optional[List[float]]
 
 
-class CompressorModelResult(ConsumerModelResultBase, CompressorTrainResult):
+class CompressorModelResult(ConsumerModelResultBase):
     componentType: Literal[ComponentType.COMPRESSOR]
+    # max_standard_rate: Optional[Union[List[Optional[float]], List[List[Optional[float]]]]]
+    failure_status: List[Optional[CompressorTrainCommonShaftFailureStatus]]
     requested_inlet_pressure: TimeSeriesFloat
     requested_outlet_pressure: TimeSeriesFloat
+    rate_sm3_day: TimeSeriesRate
+    stage_results: List[CompressorStageResult]
+    turbine_result: Optional[TurbineResult] = None
+    energy_usage_unit: Unit
+    power_unit: Unit
+
+    def to_time_series(self, stage_result: list, unit: Unit) -> TimeSeriesRate:
```

Comment:
@jsolaas @TeeeJay : will it help to add this function? Less repeated code in graph_result.

Reply:
In general it is a good idea, and IMO and aligned with "core should know less about details, just numbers", but the problem here is that core depends on an outer layer, so to speak. It should be other way around. Difficult to say, because it seems that some of these classes are "both core and not core". So, good idea, and we will eventually do something like that, but I do not like the direction of the dependency. However, "damage" has already been done, as this direction of dependency is already there... I think we should solve the dependency issue first..

Reply:
Ok, we wait - and do everything in get_asset_result, like suggested?